alumet/src/pipeline/control.rs

```diff
@@ -0,0 +1,213 @@
+use super::builder::elements::{
+    AutonomousSourceBuilder, ManagedSourceBuilder, ManagedSourceRegistration, SendSourceBuilder,
+};
+use super::elements::{output, source, transform};
+use super::{builder, trigger, PluginName, Source};
+use tokio::runtime;
+use tokio::sync::mpsc::{self, Receiver, Sender};
+use tokio::task::JoinHandle;
+use tokio_util::sync::CancellationToken;
+
+#[derive(Clone)]
+pub struct AnonymousControlHandle {
+    tx: Sender<ControlMessage>,
+    shutdown: CancellationToken,
+}
+
+#[derive(Clone)]
+pub struct ScopedControlHandle {
+    inner: AnonymousControlHandle,
+    plugin: PluginName,
+}
+
+pub enum ControlMessage {
+    Source(source::ControlMessage),
+    Transform(transform::ControlMessage),
+    Output(output::ControlMessage),
+}
+
+pub(crate) struct PipelineControl {
+    sources: source::SourceControl,
+    transforms: transform::TransformControl,
+    outputs: output::OutputControl,
+}
+
+pub enum ControlError {
+    ChannelFull(ControlMessage),
+    Shutdown,
+}
+
+impl AnonymousControlHandle {
+    pub async fn send(&mut self, message: ControlMessage) -> Result<(), ControlError> {
+        self.tx.send(message).await.map_err(|_| ControlError::Shutdown)
+    }
```

Comment:
`send` is useful in case you want to wait for the queue to have enough space to accept your message
`try_send` will immediately return an error if the queue is full, which you have to handle (what to do?)